include/alpaka/mem/buf/cpu/Set.hpp

```diff
@@ -67,10 +69,27 @@ namespace alpaka
                             TView & view,
                             std::uint8_t const & byte,
                             TExtent const & extent) :
-                                m_view(view),
                                 m_byte(byte),
-                                m_extent(extent)
-                        {}
+                                m_extentWidth(static_cast<Size>(extent::getWidth(extent))),
+                                m_extentHeight(static_cast<Size>(extent::getHeight(extent))),
+                                m_extentDepth(static_cast<Size>(extent::getDepth(extent))),
+                                m_dstWidth(static_cast<Size>(extent::getWidth(view))),
+                                m_dstHeight(static_cast<Size>(extent::getHeight(view))),
+#if (!defined(NDEBUG)) || (ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL)
+                                m_dstDepth(static_cast<Size>(extent::getDepth(view))),
+#endif
+                                m_extentWidthBytes(m_extentWidth * sizeof(elem::Elem<TView>)),
+                                m_dstPitchBytesX(mem::view::getPitchBytes<dim::Dim<TView>::value - 1u>(view)),
+                                m_dstPitchBytesY(mem::view::getPitchBytes<dim::Dim<TView>::value - 2u>(view)),                                
```

Comment:
Currently you are failing here beause for a 1D Buffer this is asking for the Pitch of the dimension -1.
To fix the current bug, it should be enough to specialize a copy for the 1D case.